### ISSUE TITLE
Value provided in ExpressionAttributeNames unused in expressions: keys: {#key_SK}

### ISSUE THREAD

**Issue by happylinks** [neoeinstein/modyne#7]
Hey there,
I'm running into an issue where I'm trying to query the primary index with PK: STORY#<id>, this is my code:

```

#[derive(Debug, Default)]
pub struct StoryComplete {
    story: Option<Story>,
    scenes: Vec<Scene>,
}

pub struct StoryCompleteQuery<'a> {
    story_id: &'a StoryIDRef,
}

impl QueryInput for StoryCompleteQuery<'_> {
    type Index = keys::Primary;
    type Aggregate = StoryComplete;

    fn key_condition(&self) -> expr::KeyCondition<Self::Index> {
        let partition = format!("STORY#{}", self.story_id);
        expr::KeyCondition::in_partition(partition)
    }
}

projections! {
    pub enum StoryCompleteEntities {
        Story,
        Scene,
    }
}

impl Aggregate for StoryComplete {
    type Projections = StoryCompleteEntities;

    fn merge(&mut self, item: Item) -> Result<(), Error> {
        match read_projection!(item)? {
            Self::Projections::Story(story) => self.story = Some(story),
            Self::Projections::Scene(scene) => self.scenes.push(scene),
        }
        Ok(())
    }
}

```

```
pub async fn get_story_complete(
        &self,
        story_id: &StoryID,
    ) -> Result<StoryComplete, Error> {
        let query = StoryCompleteQuery { story_id }.query();
        println!("query: {:#?}", query);
        let result = query.execute(self).await?;
        println!("result: {:#?}", result);

        let mut info = StoryComplete::default();
        info.reduce(result.items.unwrap_or_default())?;
        Ok(info)
    }
```

But for some reason it's adding key_SK to the ExpressionAttributeNames.
Debugging led me to this code: https://github.com/neoeinstein/modyne/blob/main/src/expr.rs#L181 but I don't understand it well enough to know why it's adding the Sk in this case.

Would appreciate some help!

Also, the library is really useful, thanks for the work!

**Comment by neoeinstein** [comment]
This should now be resolved by #9. I added a couple of test cases to ensure that, if you only specify a partition key, then it won't incorrectly add the sort key name to the query. Sorry for the delay in getting this fully resolved, but I had some time this week to get it completed and out.

**Comment by happylinks** [comment]
Thanks man, much appreciated! Confirmed it works!
